es-vue-base/src/lib-components/EsVerificationCode.vue

```diff
@@ -105,103 +137,86 @@ export default {
-                        // TODO: Why does chromium require setting the value directly?
-                        currentActiveElement.value = char;
-
-                        currentActiveElement = currentActiveElement.nextElementSibling;
-                        currentActiveElement?.focus();
-                    }
-                });
-            }
+            // Always emit input events; even if nothing changes
+            this.emitCodeUpdate();
         },
-        // Test pasting content
         handlePaste(event) {
-            const currentActiveElement = event.target;
-            const prevIndex = this.getElementIndex(currentActiveElement);
-
-            // TODO: Clearing this is not ideal but pasting over existing data doesn't work
-            this.code = Array(this.charCount);
-
             // Get the paste from clipboard; truncate it if its too long
             this.dataFromPaste = event.clipboardData?.getData('text')
-                .trim().split('').splice(0, this.charCount - prevIndex);
+                ?.trim()?.split('')?.splice(0, this.charCount);
 
-            // Check the data only contains accepted characters
-            const checkPasteValidity = this.checkCodeValidity(this.dataFromPaste);
-
-            // If the data doesn't pass validation don't update
-            if (!checkPasteValidity) {
-                event.preventDefault();
+            // If no data is found or clipboard api is not supported stop
+            if (!this.dataFromPaste) {
+                return;
             }
-        },
-        checkCodeValidity(code) {
-            if (!code) {
-                return false;
+
+            // Empty existing code and fill it with pasted data
+            this.code = [...this.emptyArray].map((cur, index) => this.dataFromPaste[index] || '');
+
+            // Move focus to the next empty spot or last input
+            let focusIndex = this.dataFromPaste.length;
+            if (focusIndex > this.code.length - 1) {
+                focusIndex = this.code.length - 1;
             }
+            this.$refs[`codeInput_${focusIndex}`]?.[0]?.focus();
 
-            const cleanCode = code.join('').trim();
-            const codeArray = cleanCode.split('');
+            event.preventDefault();
+            this.emitCodeUpdate();
+        },
+        // If parent value changes normalize it and trigger updates
+        valuePropChange(propValue) {
+            this.code = [...this.emptyArray].map((cur, index) => propValue[index] || '');
 
-            // Test all characters in the code one final time
-            // If a single bad character is found this will short circuit
-            const allValidChars = codeArray.every((num) => {
+            this.emitCodeUpdate(false);
+        },
+        // Emit this.code changes to keep this.value in sync
+        emitCodeUpdate(emitInput = true) {
+            const codeIsValid = this.code.every((num) => {
```

Comment:
Could this be simplified further into `this.code.every((num) => this.allowedChars.includes(num))`